keyboards/xd84/rules.mk

```
@@ -78,11 +78,10 @@ BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
 AUDIO_ENABLE = no           # Audio output on port C6
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
 HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
+LINK_TIME_OPTIMIZATION_ENABLE = yes
 
 # custom matrix setup
 CUSTOM_MATRIX = yes
-SRC = i2c_master.c custom_matrix_helper.c pca9555.c matrix.c
-
-EXTRAFLAGS += -flto
+SRC += i2c_master.c custom_matrix_helper.c pca9555.c matrix.c
```

Just a heads up, i2c and other hardware dependent things should be added to `QUANTUM_LIB_SRC`, (especially) if LTO is being used.  
```suggestion
SRC += custom_matrix_helper.c pca9555.c matrix.c
QUANTUM_LIB_SRC += i2c_master.c
```